### PR TITLE
Jasons/mca playmode tests

### PR DESCRIPTION
**Pull request by jasons-unity**
The idea behind the PageObects is to create readable tests that appear like steps in a test case.

The reason for the custom getters in the PageObject class is so that I can stop a test with "Inconclusive" results when a test step fails.  I would prefer to only fail a test when the last step (the part under test) fails.

I would like to keep the MARS and Companion tests together because as the tests progress I would like to reuse code between the two.  I will be able to filter what tests I want to execute using UTR.

I will have to modify the Yamato job still to only run the mars tests with mars and companion with companion.  Still working that out

### CODE REVIEW

File: Editor/PageObjects/Companion/LinkAccountScreenPO.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+﻿using NUnit.Framework;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+// ReSharper disable once CheckNamespace
+namespace MARSViking.Companion
+{
+    public static class LinkAccountScreenPo
```
Comment (leweyg-unity) [inline Editor/PageObjects/Companion/LinkAccountScreenPO.cs:9]: "Po" ?

Comment (jasons-unity) [reply Editor/PageObjects/Companion/LinkAccountScreenPO.cs:9]: It stands for PageObject

Comment (leweyg-unity) [reply Editor/PageObjects/Companion/LinkAccountScreenPO.cs:9]: Please expand (Unity standard is not to use acronyms)

File: Editor/PageObjects/Companion/ProjectListScreenPO.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                    Assert.Inconclusive("ProjectListScreen is missing");
+                return go;
+            }
+        }
+
+        public static Button LinkAccountButton
+        {
+            get
+            {
+                var go = GameObject.Find($"{ProjectListScreenPath}/Header/Link Account Button").GetComponent<Button>();
```
Comment (leweyg-unity) [inline Editor/PageObjects/Companion/ProjectListScreenPO.cs:28]: Consider a generic GetPageElement<T>(string path) method that does that null check, and get component in it.

File: Tests/Playmode/Companion/TestWelcomeScreen.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                {
+                    throw;
+                }
+            }
+        }
+
+        [UnityTest]
+        [Category("Use Case")]
+        [NUnit.Framework.Property("TestRailId", "649482")]
+        public IEnumerator TestCanLinkCloudAccount()
```
Comment (leweyg-unity) [inline Tests/Playmode/Companion/TestWelcomeScreen.cs:36]: Nice! (just saying, great to see those utility methods in action)

File: Editor/PageObjects/Companion/ProjectListScreenPO.cs (hunk 2 above)
Comment (leweyg-unity) [reply Editor/PageObjects/Companion/ProjectListScreenPO.cs:28]: GetPageElement < ComponentType > ( string path )

File: Editor/PageObjects/Companion/LinkAccountScreenPO.cs (hunk 1 above)
Comment (jasons-unity) [reply Editor/PageObjects/Companion/LinkAccountScreenPO.cs]: ok

File: Editor/PageObjects/Companion/ProjectListScreenPO.cs (hunk 2 above)
Comment (jasons-unity) [reply Editor/PageObjects/Companion/ProjectListScreenPO.cs]: I do not know how to write what you are suggesting in a way that maintains readability in method name as well as with error output when the element is not found.  I refactored the code a bit to be more reusable.  Let me know if that is sufficient or perhaps you could suggest something that maintains unique method names for readability and maintains clear output when the element is not found

File: Editor/PageObjects/Companion/LinkAccountScreenPageObject.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                    MethodBase.GetCurrentMethod().Name.Replace("get_", ""), 
+                    MethodBase.GetCurrentMethod().DeclaringType.Name.Replace("PageObject", "")) ? go.GetComponent<TMP_InputField>() : null;
+            }
+        }
+
+        public static Button LinkAccountButton
+        {
+            get
+            {
+                var go = GameObject.Find($"{LinkAccountScreenPath}/Scan View/Bottom ThreeRows/Row Buttons/Button Rectangle");
```
Comment (leweyg-unity) [inline Editor/PageObjects/Companion/LinkAccountScreenPageObject.cs:42]: static T FindObject<T>(string path) {
var go = GameObject.Find<T>(path);
if (go) return go;
throw new Exception($"Didn't find object at path {path} of type {typeof(T).Name}");
}

Comment (leweyg-unity) [reply Editor/PageObjects/Companion/LinkAccountScreenPageObject.cs:42]: public static Button LinkAccountButton => FindObject<Button>($"{LinkAccountScreenPath}/Scan View/Bottom ThreeRows/Row Buttons/Button Rectangle");

Comment (leweyg-unity) [reply Editor/PageObjects/Companion/LinkAccountScreenPageObject.cs:42]: public static Button LinkAccountButton => FindObject<Button>($"{LinkAccountScreenPath}/Scan View/Bottom ThreeRows/Row Buttons/Button Rectangle");

Comment (leweyg-unity) [reply Editor/PageObjects/Companion/LinkAccountScreenPageObject.cs:42]: static T FindObject(string path) {
var go = GameObject.Find(path);
if (go) return go;
throw new TestInconclusizeException($"Test itself failed: finding object at path {path} of type {typeof(T).Name}");
}

File: Editor/PageObjects/Companion/ProjectListScreenPO.cs (hunk 2 above)
Comment (jasons-unity) [reply Editor/PageObjects/Companion/ProjectListScreenPO.cs]: Refactored again.  Should be good this time

**Pull request merged**